pkg/trustedcabundle/trustedcabundle.go

```diff
@@ -27,16 +27,20 @@ const (
 	CADataFieldName = "odh-ca-bundle.crt"
 )
 
-func ShouldInjectTrustedBundle(ns client.Object) bool {
-	if !strings.HasPrefix(ns.GetName(), "openshift-") && !strings.HasPrefix(ns.GetName(), "kube-") &&
-		ns.GetName() != "default" && ns.GetName() != "openshift" && !HasCABundleAnnotationDisabled(ns) {
-		return true
-	}
-	return false
+func ShouldInjectTrustedBundle(ns *corev1.Namespace) bool {
+	isActive := ns.Status.Phase == corev1.NamespaceActive
+
+	return isActive && isNotReservedNamespace(ns) && !HasCABundleAnnotationDisabled(ns)
+}
+
+func isNotReservedNamespace(ns *corev1.Namespace) bool {
```

Comment:
should this function be moved in to cluster package cluster_config.go?

Reply:
Maybe, I just extracted it to improve readability. Can move it elsewhere as it's not necessarily specific to this package

Reply:
oh, to some package which we could potentially use in the future, i am not holding this as a blocker for this PR :D

Reply:
That's been moved now.